File: 202505/0001.Recurrsion_Fibonacci_Sequence.py
#v1



def fib(n):
    a = b = 1

    if n >= 1:
        print("The 1st fib element is:", a)
    if n >= 2:
        print("The 2nd fib element is:", b)

    for i in range(3, n + 1):
        temp = a + b
        print(f'The {i}th fib element is:', temp)
        a = b
        b = temp





# Example usage:
#fib(10)






#v2







def main():
    a, b = 1, 1

    n = int(input("Enter how many numbers to print: "))

    # Print the first two numbers
    if n >= 1:
        print(a)
    if n >= 2:
        print(b)

    # Print the rest
    for i in range(3, n + 1):
        c = a + b
        print(c)
        a, b = b, c

if __name__ == "__main__":
    main()

File: 202505/0001.Recurrsion_Fibonacci_Sequence.c
#include <stdio.h>

main() {
  int a;
  int b;
  int c;
  int i; 

  a = 1;
  b = 1;

  for (i = 0; i < 10; i++) {
    c = a + b;
    printf("%d\n", c); 
    a = b + c;
    printf("%d\n", a);
    b = c + a;
    printf("%d\n", b);
  }

  return 0; 
}







#include <stdio.h>

int main() {
    int a = 1, b = 1, c;
    int n, i;

    printf("Enter how many numbers to print: ");
    scanf("%d", &n);

    // Print the first two numbers
    if (n >= 1) printf("%d\n", a);
    if (n >= 2) printf("%d\n", b);

    // Print the rest
    for (i = 3; i <= n; i++) {
        c = a + b;
        printf("%d\n", c);
        a = b;
        b = c;
    }

    return 0;
}
